### PR TITLE
feat(react): support `ReactLynx::hooks::setState` trace for function components

### PR DESCRIPTION
**Pull request by upupming**
## Summary by CodeRabbit

* **New Features**
  * Added runtime tracing that captures hook and setState value diffs during profiling to improve visibility into state changes.

* **Bug Fixes**
  * Fixed profiling guard to avoid runtime errors when profiling is not defined.

* **Chores**
  * Extended public runtime/type surface to support hook-state tracking and tracing.

* **Tests**
  * Added comprehensive tests validating hook/state profiling behavior and edge cases.

This PR addresses gaps in the profiling implementation for functional components in ReactLynx and ensures full test coverage.

### Changes

- **Fix Hook Interception**: Correctly access `vnode.__c.__H` to intercept hook state in functional components.
- **Constant Mismatch Fix**: Ensure `NEXT_VALUE` aligns with Preact's internal `__N` property.
- **Robustness**: Added defensive checks for missing component instances and non-array hook values.
- **Background Check**: Ensured functional component tracing only runs when `__BACKGROUND__` is true.

### Testing & Coverage

- Added `__test__/debug/trace.test.jsx` consolidating all functionality and coverage tests.
- **Capabilities Verified**:
  - `useState` tracing (updates, flowId generation).
  - Function values in state.
  - Unserializable values (circular refs).
  - Edge cases: manual vnode construction, non-array hooks, error handling.
- **Coverage**: Achieved **100% code coverage** for `src/debug/profile.ts`.

### Verification

Run tests:

```bash
pnpm test
```

### CODE REVIEW

File: packages/react/runtime/src/debug/profile.ts [hunk 1]
```diff
@@ -106,6 +150,54 @@ export function initProfileHook(): void {
...
+            // define a setter for __N to track the next value of the hook
+            Object.defineProperty(hookState, NEXT_VALUE, {
+              get: () => hookState['internalNextValue'],
+              set: (value) => {
+                if (Array.isArray(value)) {
+                  // hookState[VALUE] is [state, dispatch]
+                  const currentValueTuple = hookState[VALUE] as unknown[];
+                  const currentValue = currentValueTuple[0];
+                  const [nextValue] = value as unknown[];
+
```
[the remark this answers is not in the document]
Comment (upupming) [reply packages/react/runtime/src/debug/profile.ts:174]: Only `useState` hooks will set `_nextValue` of a hook state, and `useState` hooks will always be an array. If it is not an array, there are bugs of our framework, so failing is a better choice.

File: packages/react/runtime/src/debug/profile.ts [hunk 2]
```diff
@@ -106,6 +155,52 @@ export function initProfileHook(): void {
     });
 
     hook(options, DIFFED, (old, vnode) => {
+      if (typeof __BACKGROUND__ !== 'undefined' && __BACKGROUND__) {
+        const hooks = vnode[COMPONENT]?.[HOOKS];
+        const hookList = hooks?.[LIST];
+
+        if (Array.isArray(hookList)) {
+          hookList.forEach((hookState, hookIdx: number) => {
+            hookState['internalNextValue'] = hookState[NEXT_VALUE];
```
Comment (HuJean) [inline packages/react/runtime/src/debug/profile.ts:164]: Is this a key consumed internally by preact?

Comment (upupming) [reply packages/react/runtime/src/debug/profile.ts:164]: No, it is only consumed by us as a tmp value.
